### PR TITLE
[msal-common] Fix caching for multiple resources

### PR DESCRIPTION
**Pull request by tnorling**
`offline_access` scope was causing all scope sets to intersect, meaning all access tokens were being cached under the same entry with new scopes appended in the cache key. This PR aims to address this by ignoring `offline_access` when attempting to determine if we are adding or updating an access token in the cache.

This will fix: #1487

### CODE REVIEW

**Comment by DarylThayil** [comment]
In general looks good can we add a test to validate different scope scenarios ?

File: lib/msal-common/src/auth/ScopeSet.ts [hunk 1]
```diff
@@ -147,13 +147,21 @@ export class ScopeSet {
 
     /**
      * Check if scopes intersect between this set and another.
+     * Do not allow offline_access to be the only intersecting scope
```
Comment (jasonnutter) [inline lib/msal-common/src/auth/ScopeSet.ts:150]: I think this comment should be below on 159, along with any other nuances you think someone should know.

File: lib/msal-common/src/auth/ScopeSet.ts [hunk 2]
```diff
@@ -153,7 +153,15 @@ export class ScopeSet {
...
-        return this.unionScopeSets(otherScopes).size < (this.scopes.size + otherScopes.getScopeCount());
+
+        const unionScopes = this.unionScopeSets(otherScopes);
+
+        // Do not allow offline_access to be the only intersecting scope
+        const sizeOtherScopes = otherScopes.containsScope(Constants.OFFLINE_ACCESS_SCOPE)? otherScopes.getScopeCount() - 1 : otherScopes.getScopeCount();
+        const sizeThisScopes = this.containsScope(Constants.OFFLINE_ACCESS_SCOPE)? this.getScopeCount() - 1: this.getScopeCount();
+        const sizeUnionScopes = unionScopes.has(Constants.OFFLINE_ACCESS_SCOPE)? unionScopes.size - 1: unionScopes.size;
+
+        return sizeUnionScopes < (sizeThisScopes + sizeOtherScopes);
```
Comment (sameerag) [inline lib/msal-common/src/auth/ScopeSet.ts:164]: Dumb q: Why are checking count than content? What if count matches and content differs for the set? I may be missing some logic and will check. This is mostly a question I immediately had.

Comment (tnorling) [reply lib/msal-common/src/auth/ScopeSet.ts:164]: `unionScopes` is a set of scope array 1 + scope array 2 containing only unique entries. If the size of this is less than the size of both arrays added together then you know there were shared scopes between the 2 individual arrays

Comment (pkanher617) [reply lib/msal-common/src/auth/ScopeSet.ts:164]: Size of both sets, not arrays. these are all unique values.

Comment (sameerag) [reply lib/msal-common/src/auth/ScopeSet.ts]: okay, got it.

**Pull request merged**